### ISSUE TITLE
Unknown js error after clicking on 'CREATE FOLDER IN DMS' button 

### ISSUE THREAD

**Issue by ghost** [acsone/alfodoo#116]
Odoo version : Odoo 11.0+e (Enterprise Edition)
Alfodoo version :  11.0.4.0.0

Following the example from the guide

model.py
```
from odoo import models
from odoo.addons.cmis_field import fields


class CrmClaim(models.Model):
    _inherit = 'crm.claim'

    cmis_folder = fields.CmisFolder()
```

view.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<odoo>
    <record id="crm_case_claims_form_view" model="ir.ui.view">
        <field name="name">CRM - Claims Form (cmis_crm_claim)</field>
        <field name="model">crm.claim</field>
        <field name="inherit_id" ref="crm_claim.crm_case_claims_form_view"/>
        <field name="arch" type="xml">
            <notebook position="inside">
                <page string="Documents" groups="base.group_user">
                    <field name="cmis_folder"/>
                </page>
            </notebook>
        </field>
    </record>
</odoo>
```
Clicking on button below
![button](https://user-images.githubusercontent.com/44695245/47902436-6e129500-de58-11e8-9ba4-7bcb9c1688a3.png)

generates this error but everything still works, including the creation of the folder
```
Error:
Uncaught TypeError: Cannot read property 'type' of undefined

http://localhost/web/static/src/js/views/basic/basic_model.js:1285
Traceback:
TypeError: Cannot read property 'type' of undefined
    at Class._applyChange (http://localhost/web/static/src/js/views/basic/basic_model.js:1285:23)
    at Object.<anonymous> (http://localhost/web/static/src/js/core/concurrency.js:216:31)
    at Object.<anonymous> (http://localhost/web/static/lib/jquery/jquery.js:3276:89)
    at fire (http://localhost/web/static/lib/jquery/jquery.js:3119:58)
    at Object.add [as done] (http://localhost/web/static/lib/jquery/jquery.js:3165:49)
    at Array.<anonymous> (http://localhost/web/static/lib/jquery/jquery.js:3275:77)
    at Function.each (http://localhost/web/static/lib/jquery/jquery.js:383:58)
    at Object.<anonymous> (http://localhost/web/static/lib/jquery/jquery.js:3272:56)
    at Function.Deferred (http://localhost/web/static/lib/jquery/jquery.js:3332:30)
    at Object.then (http://localhost/web/static/lib/jquery/jquery.js:3271:55)
```

**Comment by lmignon** [comment]
@ErpSoftAppNG Thank you for the report. Can you check into the odoo log if there is no stacktrace generated when you click on the create button on the server side. I'm not able to reproduce this bug on my installation.

**Comment by ghost** [comment]
i created a new setup on a fresh system, i still got the same error. Something is wrong somewhere ?

**Comment by lmignon** [comment]
@ErpSoftAppNG I am quite disappointed because I can't reproduce the problem on my side. If everything works despite the error in javascript, it is possible that the problem comes from a deffered method in javascript that is launched after the directory is created.  It would be necessary to do some javascript debugging....

"crm.claim" no more exists into odoo11, I suppose that the provided code into the description is not the right one...

**Comment by ghost** [comment]
Yea sure, i just copied  the example in the doc here since the error seems to be general with all models.  

The errors comes from
```
 /**
     * This is an extremely important method.  All changes in any field go
     * through this method.  It will then apply them in the local state, check
     * if onchanges needs to be applied, actually do them if necessary, then
     * resolves with the list of changed fields.
     *
     * @param {string} record_id
     * @param {Object} changes a map field => new value
     * @param {Object} [options] will be transferred to the applyChange method
     *   @see _applyChange
     * @returns {string[]} list of changed fields
     */
    notifyChanges: function (record_id, changes, options) {
        return this.mutex.exec(this._applyChange.bind(this, record_id, changes, options));
    },
```


When this method is called 
`changes = {value : undefined }`
Which is suppose to be 
`changes = { <name_of_field> : <new_value> }`
This loop triggers the error  since there is no field called `value`
```
for (var fieldName in changes) {
            field = record.fields[fieldName];
            if (field.type === 'one2many' || field.type === 'many2many') {
                defs.push(this._applyX2ManyChange(record, fieldName, changes[fieldName], options.viewType));
            } else if (field.type === 'many2one' || field.type === 'reference') {
                defs.push(this._applyX2OneChange(record, fieldName, changes[fieldName]));
            } else {
                record._changes[fieldName] = changes[fieldName];
            }
        }
```
**Currently the solution is to change the field name to value**
```
from odoo import models
from odoo.addons.cmis_field import fields


class CrmClaim(models.Model):
    _inherit = 'crm.claim'

    value = fields.CmisFolder()

```

so It just passes through smoothly

**Comment by lmignon** [comment]
@ErpSoftAppNG Thank you for the report.  https://github.com/acsone/alfodoo/commit/a3fb397cfdde9d495052505c7b448e1d140a6c76 should solve the issue. 
The fix is available on the branch https://github.com/acsone/alfodoo/tree/11.0-dev

Can you confirm that the issue is now solved?

Regards,

lmi

**Comment by ghost** [comment]
Yes issue is solved on 11.0-dev. Thanks

**Comment by lmignon** [comment]
Thank you for the feedback @ErpSoftAppNG
